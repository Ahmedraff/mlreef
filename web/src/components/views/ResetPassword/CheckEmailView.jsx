import React from 'react';
import PropTypes from 'prop-types';
import MButton from 'components/ui/MButton';

const CheckEmailView = ({ history }) => (
  <div className="reset-view_form_content">
    <div>
      <p className="text-highlight">
        Check your email for a link
        to reset your password. If it
        doesn't appear within a few
        minutes, check your spam folder.
      </p>
      <div className="reset-view_submit">
        <MButton onClick={() => { history.push('/'); }} type="button" className="btn btn-primary">
          Return to sign in
        </MButton>
      </div>
    </div>
  </div>
);

CheckEmailView.defaultProps = {
  history: {},
};

CheckEmailView.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func,
  }),
};

export default CheckEmailView;
