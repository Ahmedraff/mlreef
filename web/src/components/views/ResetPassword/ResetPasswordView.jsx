import React, { useState } from 'react';
import MInput from 'components/ui/MInput';
import MButton from 'components/ui/MButton';
import { toastr } from 'react-redux-toastr';
import './ResetPassword.scss';
import PropTypes from 'prop-types';
import PasswordManagementApi from 'apis/PasswordManagementApi';
import CheckEmailView from './CheckEmailView';
import icon from '../../../images/MLReef_Icon_POS_BETA-01.png';

const passwordApi = new PasswordManagementApi();

const ResetPasswordView = (props) => {
  const { history } = props;
  const [flag, setFlag] = useState(0);
  const [email, setEmail] = useState('');

  const sendResetEmail = (e) => {
    e.preventDefault();

    passwordApi.sendResetPassEmail(email)
      .then(() => {
        setFlag(1);
        toastr.info('Email sent', 'A Reset password link has been sent to your email');
      })
      .catch(() => {
        if (email !== '') toastr.error('Error', 'Sorry!! your email does not exist with us');
        else toastr.error('Error', 'Please enter a valid Email Id');
      });
  };

  return (
    <div className="reset-password">
      <div className="reset-password_container">
        <div className="reset-password_brand">
          <img className="brand" src={icon} alt="" />
        </div>
        <form className="reset-view_form">
          <div className="reset-view_form_header">
            Reset password for MLReef
          </div>
          {flag === 0
            ? (
              <>
                <div className="reset-view_form_content">
                  <div className="cols-2">
                    <MInput
                      id="email"
                      value={email}
                      label="Enter your email address"
                      onChange={(e) => setEmail(e.target.value)}
                    />
                  </div>
                </div>
                <div className="reset-view_submit">
                  <MButton onClick={sendResetEmail} id="password-submit" type="submit" className="btn btn-primary">
                    Send password reset email
                  </MButton>
                </div>
              </>
            )
            : <CheckEmailView history={history} />}
        </form>
        <footer className="reset-view_form_footer">
          {'New to MLReef? '}
          <a href="/register" className="btn-link"><b>Create an account</b></a>
        </footer>
      </div>
    </div>
  );
};

ResetPasswordView.defaultProps = {
  history: {},
};

ResetPasswordView.propTypes = {
  history: PropTypes.shape({}),
};

export default ResetPasswordView;
