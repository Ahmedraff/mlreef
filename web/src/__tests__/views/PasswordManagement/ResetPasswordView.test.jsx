import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import ResetPasswordView from 'components/views/ResetPassword/ResetPasswordView';
import CheckEmailView from 'components/views/ResetPassword/CheckEmailView';

const props = {
  history: {
    push: jest.fn(),
  },
};

describe('presence of basic user elements', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ResetPasswordView props={props} />);
  });

  test('assert that snapshot matches', () => {
    const snapShot = renderer.create(<ResetPasswordView props={props} />).toJSON();
    expect(snapShot).toMatchSnapshot();
  });

  test('assert password reset button changes state', () => {
    wrapper.find('#password-submit').simulate('click');
    expect(wrapper.find('CheckEmailView')).toHaveLength(1);
  });
});
