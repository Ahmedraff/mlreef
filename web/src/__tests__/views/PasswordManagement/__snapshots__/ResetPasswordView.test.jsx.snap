// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presence of basic user elements assert that snapshot matches 1`] = `
<div
  className="reset-password"
>
  <div
    className="reset-password_container"
  >
    <div
      className="reset-password_brand"
    >
      <img
        alt=""
        className="brand"
        src={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "process": [Function],
          }
        }
      />
    </div>
    <form
      className="reset-view_form"
    >
      <div
        className="reset-view_form_header"
      >
        Reset password for MLReef
      </div>
      <div
        className="reset-view_form_content"
      >
        <div
          className="cols-2"
        >
          <div
            className="m-input "
          >
            <div
              className="m-input_container"
            >
              <label
                htmlFor="email"
              >
                Enter your email address
              </label>
              <input
                className="m-input_input "
                id="email"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder={null}
                readOnly={false}
                type="text"
                value=""
              />
              <div
                className="m-input_errors"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="reset-view_submit"
      >
        <button
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="submit"
        >
          Send password reset email
        </button>
      </div>
    </form>
    <footer
      className="reset-view_form_footer"
    >
      New to MLReef? 
      <a
        className="btn-link"
        href="/register"
      >
        <b>
          Create an account
        </b>
      </a>
    </footer>
  </div>
</div>
`;
